ThirdPartyAdapters/unity/unity/src/main/java/com/google/ads/mediation/unity/UnityAdapter.java

```
@@ -119,31 +121,69 @@ public void onUnityAdsClick(String s) {
         public void onUnityAdsPlacementStateChanged(String placementId,
                                                     UnityAds.PlacementState oldState,
                                                     UnityAds.PlacementState newState) {
-            // This callback is not forwarded to the adapter by the UnitySingleton and the
-            // adapter should use the onUnityAdsReady and onUnityAdsError callbacks to forward
-            // Unity Ads SDK state to Google Mobile Ads SDK.
+            // Unity Ads SDK NO_FILL state to Google Mobile Ads SDK.
+            if (mMediationInterstitialListener != null && placementId.equals(getPlacementId()) && newState.equals(UnityAds.PlacementState.NO_FILL)) {
+                Log.e(TAG, "UnityAds no fill: " + placementId);
+                mMediationInterstitialListener.onAdFailedToLoad(UnityAdapter.this, AdRequest.ERROR_CODE_NO_FILL);
+            }
         }
 
         @Override
         public void onUnityAdsFinish(String placementId, UnityAds.FinishState finishState) {
             // Unity Ads ad closed.
-            if (mMediationInterstitialListener != null) {
-                mMediationInterstitialListener.onAdClosed(UnityAdapter.this);
+            if (placementId.equals(getPlacementId())) {
+                if (mMediationInterstitialListener != null) {
+                    if (finishState == UnityAds.FinishState.ERROR) {
+                        mMediationInterstitialListener.onAdFailedToLoad(UnityAdapter.this, AdRequest.ERROR_CODE_INTERNAL_ERROR);
```

according to docs, `mMediationInterstitialListener.onAdFailedToLoad` should be used only when we failed to load ad.

---

It looks like we should just ignore error here and do nothing. Also we should check with AdMob team.